### PR TITLE
Document ASTNode#symbolize

### PR DESCRIPTION
**Pull request by Sija**
SSIA, couldn't find it in the docs so went ahead and fixed it up :)

### CODE REVIEW

File: src/compiler/crystal/macros.cr [hunk 1]
```diff
@@ -153,6 +153,18 @@ module Crystal::Macros
...
     end
 
+    # Returns a `SymbolLiteral` that contains this node's textual representation.
+    #
+    # ```
+    # macro test
+    #   {{ "foo".symbolize }}
+    # end
+    #
+    # puts test # prints "foo" (including the double quotes)
```
Comment (bcardiff) [inline src/compiler/crystal/macros.cr]: Maybe is cleaner to show `{{ "foo".symbolize }} == :"\"foo\""  # => true`. 

I am not 100% why the double quotes are included in the symbol value though.

Comment (Sija) [reply src/compiler/crystal/macros.cr]: I was wondering the same :) @asterite, would you please tell us why❓

Comment (asterite) [reply src/compiler/crystal/macros.cr]: The textual representation of "foo" is "foo", and so a symbol with "foo" is `:"\"foo\"`. If you want `:foo` you need to do `"foo".id.symbolize`.

Comment (asterite) [reply src/compiler/crystal/macros.cr]: Also `"foo".stringify` makes a string literal out of that, so it becomes `"\"foo\""`. The two methods are similar.

**Comment by Sija** [comment]
@bcardiff updated code example to match eye expectations (though usage of `.id` might seem cryptic)

**Pull request merged**